code/game/gamemodes/wizard/spellbook.dm

```diff
@@ -57,20 +61,58 @@
 
 	available_artifacts = typesof(/datum/spellbook_artifact) - /datum/spellbook_artifact
 
+
+//I tried turning this into a proc and all it did was give me a headache and waste half an hour of my life, so here you go
 	for(var/type_S in getAllWizSpells())
```

Comment:
There is absolutely no need to have 5 `for` loops in all wizards spells and instantiate said spells 5 times
Make a single `for` loops and assign spells to their proper categories in a single go